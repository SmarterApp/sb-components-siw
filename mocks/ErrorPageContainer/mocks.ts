import { pageType, ErrorPageContainerProps } from "@src/index";

export const serverErrMock: ErrorPageContainerProps = {
  errorCode: pageType.ServerError,
  errorMsg: "you did a bad thing that caused errors",
  description: "stop doing that thing that caused errors"
};

export const notFoundErrMock: ErrorPageContainerProps = {
  errorCode: pageType.NotFound,
  errorMsg: "This item cannot be found",
  description: "Please try another search"
};
