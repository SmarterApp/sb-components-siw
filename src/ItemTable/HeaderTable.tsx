import * as React from "react";
import {
  SortColumnModel,
  HeaderSortModel,
  SortDirection,
  ColumnGroup
} from "@src/index";

/**
 * Properties for HeaderTable component
 * @interface HeaderTableProps
 */
export interface HeaderTableProps {
  columns: ColumnGroup[];
  onHeaderClick: (header: ColumnGroup) => void;
  sorts: HeaderSortModel[];
  isLinkTable: boolean;
}

const invokeResetSortLimit = 2;

const style = {
  color: "gray"
};

const descendingArrow = (
  <span
    style={style}
    className="fa fa-sort-desc sort-icon"
    aria-hidden="true"
  />
);

const ascendingArrow = (
  <span style={style} className="fa fa-sort-asc sort-icon" aria-hidden="true" />
);

const noSort = (
  <span style={style} className="fa fa-sort sort-icon" aria-hidden="true" />
);

/**
 * HeaderTable creates a table header based on the passed in columns
 * The HeaderTable, when clicked, will add the clicked column header
 * the parent, ItemTableContainer, state, sorting the table
 */
export class HeaderTable extends React.Component<HeaderTableProps, {}> {
  constructor(props: HeaderTableProps) {
    super(props);
  }
  /**
   * Send the clicked sort column to the parent to be added to the sorts list
   * using onHeaderClick prop.
   * @param {ColumnGroup} sCol
   * @param {(HeaderSortModel | undefined)} hCol
   */
  headerClickHandler = (sCol: ColumnGroup, hCol?: HeaderSortModel) => {
    this.props.onHeaderClick(sCol);
  };

  /**
   * Send the clicked sort column to the parent to be added to the sorts list
   * using onHeaderClick prop.
   * @param {React.KeyboardEvent<any>} e
   * @param {HeaderColumnModel} sCol
   * @param {(HeaderSortModel | undefined)} hCol
   */
  headerKeyUpHandler = (
    e: React.KeyboardEvent<HTMLTableHeaderCellElement>,
    sCol: ColumnGroup,
    hCol?: HeaderSortModel
  ) => {
    if (e.keyCode === 13) {
      this.props.onHeaderClick(sCol);
    }
  };

  /**
   * Assigns an ascending or descending arrow character to the visible column
   * header when clicked, denoting how it is sorted
   * @param {(HeaderSortModel | undefined)} headerSort
   * @returns {JSX.Element}
   */
  setDirElem(headerSort: HeaderSortModel | undefined): JSX.Element {
    let dirElem = noSort;
    if (!headerSort) {
      dirElem = noSort;
    } else if (headerSort.direction === SortDirection.Ascending) {
      dirElem = ascendingArrow;
    } else if (headerSort.direction === SortDirection.Descending) {
      dirElem = descendingArrow;
    }

    return dirElem;
  }

  /**
   * Renders a single table header element and the corresponding ascending, descending,
   * or 'not-sorted' symbol
   * @param {ColumnGroup} col
   * @returns {JSX.Element}
   */
  renderHeader(col: ColumnGroup): JSX.Element {
    const headerSort = this.props.sorts.find(
      hs => hs.col.header === col.header
    );
    if (headerSort) {
      this.setDirElem(headerSort);
    }

    return (
      <th
        key={col.header}
        className={col.headerClassName}
        onClick={() => this.headerClickHandler(col, headerSort)}
        onKeyUp={e => this.headerKeyUpHandler(e, col, headerSort)}
        tabIndex={0}
        scope="col"
      >
        <div className={col.headerClassName}>
          <span>{col.header}</span>
          {this.setDirElem(headerSort)}
        </div>
      </th>
    );
  }

  render() {
    return (
      <thead>
        <tr className="primary">
          {this.props.isLinkTable ? (
            undefined
          ) : (
            <td className="td-header" colSpan={1} />
          )}
          {this.props.columns.map(col => this.renderHeader(col))}
        </tr>
      </thead>
    );
  }
}
