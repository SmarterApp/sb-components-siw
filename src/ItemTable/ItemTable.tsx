import * as React from "react";
import {
  getResourceContent,
  ItemCard,
  ItemTableRow,
  ToolTip,
  ItemCardViewer,
  ItemCardModel,
  itemIdEqual,
  AboutItemModel,
  Resource,
  HeaderSortModel,
  SortColumnModel,
  ColumnGroup
} from "@src/index";
import { TestCodeToLabel } from "@src/ItemSearch/ItemSearchModels";

export interface ItemTableProps {
  cardRows: ItemCardModel[];
  onRowExpand: (item: ItemCardModel) => void;
  onRowSelect: (item: ItemCardModel) => void;
  sort: HeaderSortModel[];
  columns: ColumnGroup[];
  expandedRow?: ItemCardModel;
  item?: Resource<AboutItemModel>;
  isLinkTable: boolean;
  isItemSelected: boolean;
  numberOfSelectedItem: number;
  getSelectedItemCount: () => number;
  showErrorModalOnPrintItemsCountExceeded: () => void;
  associatedItems: any[];
  countNumberOfItemsAfterSelection: (
    currentItems: ItemCardModel[],
    selectedItemsCount: number
  ) => number;
  isInterimSite: boolean;
  testCodeToLabelMap: TestCodeToLabel;
}
/**
 * Renders the table populated from an array of ItemCardModels. Also renders an instance of the ItemCardViewer,
 * inserting a responsive sub-table with an iframe that displays the Item Card.
 * @class ItemTable
 * @extends {React.Component<ItemTableProps, {}>}
 */
export class ItemTable extends React.Component<ItemTableProps, {}> {
  constructor(props: ItemTableProps) {
    super(props);
  }

  renderAllRows(): JSX.Element[] {
    const { cardRows } = this.props;
    let rows: JSX.Element[] = [];
    cardRows.forEach(item => {
      rows = rows.concat(this.renderRow(item));
    });

    return rows;
  }

  renderExpandedRow(item: Resource<AboutItemModel>): JSX.Element | undefined {
    let result: JSX.Element | undefined;
    const itemContent = getResourceContent(item);
    if (itemContent) {
      result = (
        <tr key="item-card-viewer">
          <td colSpan={7}>
            <ItemCardViewer item={itemContent} />
          </td>
        </tr>
      );
    }

    return result;
  }

  renderRow(rowData: ItemCardModel): JSX.Element[] {
    const {
      expandedRow,
      columns,
      item,
      isLinkTable,
      onRowExpand,
      onRowSelect
    } = this.props;
    const rows: JSX.Element[] = [];
    const isExpanded =
      expandedRow && itemIdEqual(expandedRow, rowData) ? true : false;

    rows.push(
      <ItemTableRow
        key={`${rowData.bankKey}-${rowData.itemKey}`}
        rowData={rowData}
        hasControls={!isLinkTable}
        columns={columns}
        isExpanded={isExpanded}
        onRowExpand={onRowExpand}
        onRowSelect={onRowSelect}
        isItemSelected={this.props.isItemSelected}
        numberOfSelectedItem={this.props.numberOfSelectedItem}
        getSelectedItemCount={this.props.getSelectedItemCount}
        showErrorModalOnPrintItemsCountExceeded={
          this.props.showErrorModalOnPrintItemsCountExceeded
        }
        associatedItems={this.props.associatedItems}
        countNumberOfItemsAfterSelection={
          this.props.countNumberOfItemsAfterSelection
        }
        isInterimSite={this.props.isInterimSite}
        testCodeToLabelMap={this.props.testCodeToLabelMap}
      />
    );

    if (isExpanded && item) {
      const expandContent = this.renderExpandedRow(item);
      if (expandContent) {
        rows.push(expandContent);
      }
    }

    return rows;
  }

  render() {
    const { cardRows } = this.props;
    let content = <div>No Items.</div>;
    if (cardRows) {
      content = <tbody>{this.renderAllRows()}</tbody>;
    }

    return content;
  }
}
