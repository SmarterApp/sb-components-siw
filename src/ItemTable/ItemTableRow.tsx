import * as React from "react";
import {
  ItemCardModel,
  ToolTip,
  generateTooltip,
  SortColumnModel,
  ColumnGroup
} from "@src/index";
import { TestCodeToLabel } from "@src/ItemSearch/ItemSearchModels";
import {
  mapItemGrade,
  mapItemSubjectlabel,
  mapItemClaim
} from "@src/PrintCart/PrintCartItemTableRow";
import { ItemColumnHeadersConfig } from "@src/SearchResultContainer/SearchResultModels";
import { AnswerKeysRubricModal } from "@src/AnswerKeysRubrics/AnswerKeysRubricModal";

export interface ItemTableRowProps {
  rowData: ItemCardModel;
  hasControls: boolean;
  columns: ColumnGroup[];
  isExpanded: boolean;
  onRowExpand: (item: ItemCardModel) => void;
  onRowSelect: (item: ItemCardModel) => void;
  isItemSelected: boolean;
  numberOfSelectedItem: number;
  getSelectedItemCount: () => number;
  showErrorModalOnPrintItemsCountExceeded: () => void;
  associatedItems: any[];
  countNumberOfItemsAfterSelection: (
    currentItems: ItemCardModel[],
    selectedItemsCount: number
  ) => number;
  isInterimSite: boolean;
  testCodeToLabelMap: TestCodeToLabel;
  itemTableConfig: ItemColumnHeadersConfig[];
}

export interface ItemTableRowState {
  showAnswerKeysModal: boolean;
}

const unChecked = (
  <i className="fa fa-square-o fa-sm table-icon" aria-hidden="true" />
);
const checked = (
  <i className="fa fa-check-square-o fa-sm table-icon" aria-hidden="true" />
);
const collapse = (
  <i className="fa fa-chevron-right fa-sm table-icon" aria-hidden="true" />
);
const expand = (
  <i className="fa fa-chevron-down fa-sm table-icon" aria-hidden="true" />
);

export class ItemTableRow extends React.Component<
  ItemTableRowProps,
  ItemTableRowState
> {
  constructor(props: ItemTableRowProps) {
    super(props);
    this.state = {
      showAnswerKeysModal: false
    };
  }

  shouldComponentUpdate(nextProps: ItemTableRowProps, nextState: {}) {
    // return (
    //   this.props.isExpanded !== nextProps.isExpanded ||
    //   this.props.rowData.selected !== nextProps.rowData.selected ||
    //   this.props.isItemSelected !== nextProps.isItemSelected ||
    //   this.props.numberOfSelectedItem !== nextProps.numberOfSelectedItem
    // );
    return true;
  }

  //keyboard accessibility handler tooltip: select item on key press on tooltip
  handleTooltipKeyPress = (e: any) => {
    const item = this.props.rowData;
    this.handleCheckboxClick(e, item, this.shouldRowButtonBeDisabled());
  };

  handleRowClick = (rowData: ItemCardModel) => {
    this.props.onRowExpand(rowData);
  };

  handleKeyUpEnter = (e: React.KeyboardEvent<HTMLTableRowElement>) => {
    const rowData = this.props.rowData;
    if (e.keyCode === 13) {
      this.props.onRowExpand(rowData);
    }
  };

  //disabled button for an item if it associated PT item is already selected
  shouldRowButtonBeDisabled = () => {
    if (
      this.props.rowData.selected === undefined ||
      this.props.rowData.selected === false
    ) {
      if (this.props.associatedItems !== undefined) {
        const associatedItems = this.props.associatedItems;
        const itemKey = this.props.rowData.itemKey;
        for (const itemKeyInAssociatedItems in associatedItems) {
          const associatedItemsArray =
            associatedItems[itemKeyInAssociatedItems];
          for (let i = 0; i < associatedItemsArray.length; i++) {
            if (associatedItemsArray[i][0].itemKey === itemKey)
              return "disabled";
          }
        }
      }
    }

    return " ";
  };

  /**
   * @param e
   * @param rowData
   * @param shouldBeDisabled
   * Select/deselect item on checkbox click
   * Dot not perform any operation if item is PT item and any of its assocaited item is already selected
   */
  handleCheckboxClick = (
    e: React.MouseEvent<HTMLTableDataCellElement>,
    rowData: ItemCardModel,
    shouldBeDisabled: string
  ) => {
    let currentItems: ItemCardModel[] = [];
    currentItems[0] = rowData;
    e.stopPropagation();
    if (
      shouldBeDisabled == "disabled" &&
      (rowData.selected === undefined || rowData.selected === false)
    ) {
      return;
    }
    let selectedItemsCount = this.props.getSelectedItemCount();
    if (
      rowData.selected !== true &&
      this.props.countNumberOfItemsAfterSelection(
        currentItems,
        selectedItemsCount
      ) > 50
    ) {
      this.props.showErrorModalOnPrintItemsCountExceeded();
      return;
    } else {
      if (rowData.selected === true) rowData.selected = false;
      else rowData.selected = true;
      this.props.onRowSelect(rowData);
      e.stopPropagation();
    }
  };

  renderColumnGroup(
    colGroup: ColumnGroup,
    cellData: ItemCardModel
  ): JSX.Element {
    const colValues = colGroup.cols.map(c => this.renderCell(c, cellData));
    const { headerClassName } = colGroup;

    let isHidden = false;
    const tableHeaderConfig = this.props.itemTableConfig.find(
      hs => hs.headerName.toUpperCase() === colGroup.header.toUpperCase()
    );

    if (tableHeaderConfig != undefined) {
      isHidden = tableHeaderConfig.isHidden;
    }

    if (true) {
      return (
        <>
          {!isHidden && (
            <td
              key={`${headerClassName}-${cellData.bankKey}-${cellData.itemKey}`}
              className={headerClassName}
              role="gridcell"
            >
              {colValues}
            </td>
          )}
        </>
      );
    } else {
    }
  }

  closeAnswerKeysModal = () => {
    this.setState({ showAnswerKeysModal: false });
  };

  openAnswerKeyModal = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.stopPropagation();
    this.setState({ showAnswerKeysModal: true });
  };

  renderCell(col: SortColumnModel, cellData: ItemCardModel): JSX.Element {
    const columnText = col.accessor(cellData);
    let content: JSX.Element | React.ReactText | undefined;

    if (col.helpText) {
      content = generateTooltip({
        helpText: <p>{col.helpText(cellData)}</p>,
        position: "bottom",
        displayIcon: true,
        className: "box",
        displayText: columnText
      });
    } else if (col.className === "item" && !this.props.hasControls) {
      content = (
        <a tabIndex={0} role="link">
          {columnText}
        </a>
      );
    } else if (col.className === "subject") {
      content = (
        <span
          className={`table-subject-highlight ${cellData.subjectCode.toLowerCase()}`}
        >
          {mapItemSubjectlabel(columnText.toString())}
        </span>
      );
    } else if (col.className === "testname") {
      let testLabel = "";
      if (columnText !== undefined) {
        testLabel = this.props.testCodeToLabelMap[columnText];
      }
      content = <span>{testLabel}</span>;
    } else if (col.className === "grade") {
      const shortGradeValue = mapItemGrade(columnText.toString());
      content = <span>{shortGradeValue}</span>;
    } else if (col.className === "answerkeys") {
      if (columnText.toString().length > 0)
        content = (
          <span tabIndex={0} aria-label={"Answer is " + columnText}>
            {columnText}
          </span>
        );
      else {
        content = (
          <button
            type="button"
            className="btn btn-default btn-raised btn-answerkeys"
            onClick={e => {
              this.openAnswerKeyModal(e);
            }}
            aria-label="Click to view answer keys or rubrics"
          >
            View
          </button>
        );
      }
    } else {
      if (col.className === "item") {
        content = (
          <a tabIndex={0} role="link">
            {columnText}
          </a>
        );
      } else {
        content = <span>{columnText}</span>;
      }
    }

    return <span key={col.className}>{content}</span>;
  }

  renderControls(): JSX.Element[] | undefined {
    const { rowData, hasControls, isExpanded } = this.props;

    const shouldBeDisabledResult = this.shouldRowButtonBeDisabled();

    const addOrRemoveIcon = () => {
      return rowData.selected === true ? "fa-check-circle" : "fa-plus-circle";
    };
    const addRemoveItemBtnCSSClass = () => {
      return rowData.selected === true ? "btn-red-border" : "btn-blue-border";
    };

    const getToolTipMsg = () => {
      if (shouldBeDisabledResult === "disabled")
        return "This is a Performance Task and must be selected as a group in a predefined sequence. PTs are designed as a complete activity to measure a studentâ€™s ability to demonstrate critical-thinking, problem-solving skills and/or complex analysis, and writing and research skills.";
      if (addOrRemoveIcon() === "fa-plus-circle")
        return "Add item to print queue";
      else return "Remove item from print queue ";
    };

    const buttonAddOrRemove = (
      <button
        type="button"
        tabIndex={-1}
        className={`btn btn  btn-item-add-remove-grid btn-sm ${addRemoveItemBtnCSSClass()}`} //${this.props.rowData.subjectCode.toLowerCase()
      >
        <i className={"fa fa-2x " + addOrRemoveIcon()} />
      </button>
    );

    const tooltip = generateTooltip({
      displayIcon: false,
      className: "btn-table-cell",
      helpText: <label>{getToolTipMsg()}</label>,
      displayText: buttonAddOrRemove,
      onKeyPress: this.handleTooltipKeyPress
    });

    let controls: JSX.Element[] | undefined;
    if (hasControls) {
      controls = [
        <td
          className="item-checkbox"
          key="checkbox-control"
          onClick={e =>
            this.handleCheckboxClick(
              e,
              rowData,
              this.shouldRowButtonBeDisabled()
            )
          }
          aria-label="Select item to print cart"
        >
          {tooltip}
        </td>
      ];
    }

    return controls;
  }

  render() {
    const { rowData, isExpanded, columns } = this.props;

    return (
      <>
        <AnswerKeysRubricModal
          showModal={this.state.showAnswerKeysModal}
          itemCard={this.props.rowData}
          closeAnswerKeysModal={this.closeAnswerKeysModal}
        />
        <tr
          key={`${rowData.bankKey}-${rowData.itemKey}`}
          className={isExpanded ? "selected" : ""}
          onClick={() => this.handleRowClick(rowData)}
          onKeyUp={e => this.handleKeyUpEnter(e)}
        >
          {this.renderControls()}
          {columns.map(col => this.renderColumnGroup(col, rowData))}
        </tr>
      </>
    );
  }
}

// for expand...  deprecitd... after addremove icon... leaving code for future refrence if required
// <td className="arrow-indicator" tabIndex={0} key="expand-control">
//   {isExpanded ? expand : collapse}
// </td>
