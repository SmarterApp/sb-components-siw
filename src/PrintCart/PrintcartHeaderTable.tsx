import * as React from "react";
import {
  SortColumnModel,
  HeaderSortModel,
  SortDirection,
  ColumnGroup
} from "@src/index";
import { ItemColumnHeadersConfig } from "@src/SearchResultContainer/SearchResultModels";

/**
 * Properties for HeaderTable component
 * @interface HeaderTableProps
 */
export interface PrintCartHeaderTableProps {
  columns: ColumnGroup[];
  onHeaderClick: (header: ColumnGroup) => void;
  sorts: HeaderSortModel[];
  isLinkTable: boolean;
  itemTableConfig: ItemColumnHeadersConfig[];
}

const invokeResetSortLimit = 2;

const style = {
  color: "gray"
};

const descendingArrow = (
  <span
    style={style}
    className="fa fa-sort-desc sort-icon"
    aria-hidden="true"
  />
);

const ascendingArrow = (
  <span style={style} className="fa fa-sort-asc sort-icon" aria-hidden="true" />
);

const noSort = (
  <span style={style} className="fa fa-sort sort-icon" aria-hidden="true" />
);

/**
 * HeaderTable creates a table header based on the passed in columns
 * The HeaderTable, when clicked, will add the clicked column header
 * the parent, PrintCartItemTableContainer, state, sorting the table
 */
export class PrintcartHeaderTable extends React.Component<
  PrintCartHeaderTableProps,
  {}
> {
  constructor(props: PrintCartHeaderTableProps) {
    super(props);
  }
  /**
   * Send the clicked sort column to the parent to be added to the sorts list
   * using onHeaderClick prop.
   * @param {ColumnGroup} sCol
   * @param {(HeaderSortModel | undefined)} hCol
   */
  headerClickHandler = (sCol: ColumnGroup, hCol?: HeaderSortModel) => {
    this.props.onHeaderClick(sCol);
  };

  /**
   * Send the clicked sort column to the parent to be added to the sorts list
   * using onHeaderClick prop.
   * @param {React.KeyboardEvent<any>} e
   * @param {HeaderColumnModel} sCol
   * @param {(HeaderSortModel | undefined)} hCol
   */
  headerKeyUpHandler = (
    e: React.KeyboardEvent<HTMLTableHeaderCellElement>,
    sCol: ColumnGroup,
    hCol?: HeaderSortModel
  ) => {
    if (e.keyCode === 13) {
      this.props.onHeaderClick(sCol);
    }
  };

  /**
   * Assigns an ascending or descending arrow character to the visible column
   * header when clicked, denoting how it is sorted
   * @param {(HeaderSortModel | undefined)} headerSort
   * @returns {JSX.Element}
   */
  setDirElem(headerSort: HeaderSortModel | undefined): JSX.Element {
    let dirElem = noSort;
    if (!headerSort) {
      dirElem = noSort;
    } else if (headerSort.direction === SortDirection.Ascending) {
      dirElem = ascendingArrow;
    } else if (headerSort.direction === SortDirection.Descending) {
      dirElem = descendingArrow;
    }

    return dirElem;
  }

  /**
   * Renders a single table header element and the corresponding ascending, descending,
   * or 'not-sorted' symbol
   * @param {ColumnGroup} col
   * @returns {JSX.Element}
   */
  renderHeader(col: ColumnGroup): JSX.Element {
    const headerSort = this.props.sorts.find(
      hs => hs.col.header === col.header
    );

    let isHidden = false;
    const tableHeaderConfig = this.props.itemTableConfig.find(
      hs => hs.headerName.toUpperCase() === col.header.toUpperCase()
    );
    if (tableHeaderConfig != undefined) {
      isHidden = tableHeaderConfig.isHidden;
    }

    if (headerSort) {
      this.setDirElem(headerSort);
    }

    return (
      <>
        {!isHidden && (
          <th
            key={col.header}
            className={col.headerClassName}
            tabIndex={0}
            scope="col"
          >
            <div className={col.headerClassName}>
              <span>{col.header}</span>
            </div>
          </th>
        )}
      </>
    );
  }

  render() {
    return (
      <thead>
        <tr className="primary">
          {this.props.isLinkTable ? undefined : <th colSpan={1} />}
          <th colSpan={1} scope="col" />
          {this.props.columns.map(col => this.renderHeader(col))}
          <th className="th-arrrow-button" colSpan={1} scope="col" />
        </tr>
      </thead>
    );
  }
}
