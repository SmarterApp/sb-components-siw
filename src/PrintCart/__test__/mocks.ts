import { ItemCardModel } from "@src/index";
import { currentId } from "async_hooks";

export const itemHandler = jest.fn((item: ItemCardModel) => {
  return undefined;
});

export const getSelectedItemCount = jest.fn(() => {
  return;
});

export const countNumberOfItemsAfterSelection = jest.fn(
  (currentItems: ItemCardModel[], selectedItemsCount: number) => {
    return;
  }
);

export const onAddOrRemoveSelectedItems = jest.fn((item: ItemCardModel) => {
  return;
});

export const onItemsReorder = jest.fn((i: number, j: number) => {
  return;
});
