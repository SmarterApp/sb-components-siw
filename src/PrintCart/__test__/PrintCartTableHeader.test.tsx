import * as React from "react";
import * as TestUtils from "react-dom/test-utils";
import { shallow, mount, render } from "enzyme";
import {
  headerColumns,
  HeaderSortModel,
  SortColumnModel,
  SortDirection,
  HeaderTable,
  HeaderTableProps
} from "@src/index";
import { getColumnsHeaderConfig_mock, tabClassNames } from "@mocks/ItemTable/mocks";
import { itemHandler } from "./mocks";
import { PrintcartHeaderTable } from "../PrintcartHeaderTable";

describe("ItemTableHeader", () => {
  const sorts: HeaderSortModel[] = [
    {
      col: headerColumns[0],
      direction: SortDirection.Ascending,
      resetSortCount: 1
    }
  ];

  const props: HeaderTableProps = {
    sorts,
    columns: headerColumns,
    onHeaderClick: itemHandler,
    isLinkTable: false,
    itemTableConfig: getColumnsHeaderConfig_mock()
  };

  const wrapper = shallow(<PrintcartHeaderTable {...props} />);
  const wrapperLinkTable = shallow(
    <PrintcartHeaderTable {...props} isLinkTable={false} />
  );

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });


  it("matches link table snapshot", () => {
    expect(wrapperLinkTable).toMatchSnapshot();
  });
});
