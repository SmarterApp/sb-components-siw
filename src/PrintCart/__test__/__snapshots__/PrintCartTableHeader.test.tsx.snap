// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTableHeader matches link table snapshot 1`] = `
<thead>
  <tr
    className="primary"
  >
    <th
      colSpan={1}
    />
    <th
      colSpan={1}
      scope="col"
    />
    <React.Fragment>
      <th
        className="item"
        key="Item"
        scope="col"
        tabIndex={0}
      >
        <div
          className="item"
        >
          <span>
            Item
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="stimulus"
        key="Stimulus ID"
        scope="col"
        tabIndex={0}
      >
        <div
          className="stimulus"
        >
          <span>
            Stimulus ID
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="item-position-in-test"
        key="Item position in test"
        scope="col"
        tabIndex={0}
      >
        <div
          className="item-position-in-test"
        >
          <span>
            Item position in test
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="subject"
        key="Subject"
        scope="col"
        tabIndex={0}
      >
        <div
          className="subject"
        >
          <span>
            Subject
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="grade"
        key="Grade"
        scope="col"
        tabIndex={0}
      >
        <div
          className="grade"
        >
          <span>
            Grade
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="testname"
        key="Test name"
        scope="col"
        tabIndex={0}
      >
        <div
          className="testname"
        >
          <span>
            Test name
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="claim"
        key="Claim"
        scope="col"
        tabIndex={0}
      >
        <div
          className="claim"
        >
          <span>
            Claim
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="Target"
        key="Target"
        scope="col"
        tabIndex={0}
      >
        <div
          className="Target"
        >
          <span>
            Target
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="standard"
        key="Standard"
        scope="col"
        tabIndex={0}
      >
        <div
          className="standard"
        >
          <span>
            Standard
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="dok"
        key="DOK"
        scope="col"
        tabIndex={0}
      >
        <div
          className="dok"
        >
          <span>
            DOK
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="difficulty"
        key="Difficulty"
        scope="col"
        tabIndex={0}
      >
        <div
          className="difficulty"
        >
          <span>
            Difficulty
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="answerkeys"
        key="Answer key"
        scope="col"
        tabIndex={0}
      >
        <div
          className="answerkeys"
        >
          <span>
            Answer key
          </span>
        </div>
      </th>
    </React.Fragment>
    <th
      className="th-arrrow-button"
      colSpan={1}
      scope="col"
    />
  </tr>
</thead>
`;

exports[`ItemTableHeader matches snapshot 1`] = `
<thead>
  <tr
    className="primary"
  >
    <th
      colSpan={1}
    />
    <th
      colSpan={1}
      scope="col"
    />
    <React.Fragment>
      <th
        className="item"
        key="Item"
        scope="col"
        tabIndex={0}
      >
        <div
          className="item"
        >
          <span>
            Item
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="stimulus"
        key="Stimulus ID"
        scope="col"
        tabIndex={0}
      >
        <div
          className="stimulus"
        >
          <span>
            Stimulus ID
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="item-position-in-test"
        key="Item position in test"
        scope="col"
        tabIndex={0}
      >
        <div
          className="item-position-in-test"
        >
          <span>
            Item position in test
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="subject"
        key="Subject"
        scope="col"
        tabIndex={0}
      >
        <div
          className="subject"
        >
          <span>
            Subject
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="grade"
        key="Grade"
        scope="col"
        tabIndex={0}
      >
        <div
          className="grade"
        >
          <span>
            Grade
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="testname"
        key="Test name"
        scope="col"
        tabIndex={0}
      >
        <div
          className="testname"
        >
          <span>
            Test name
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="claim"
        key="Claim"
        scope="col"
        tabIndex={0}
      >
        <div
          className="claim"
        >
          <span>
            Claim
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="Target"
        key="Target"
        scope="col"
        tabIndex={0}
      >
        <div
          className="Target"
        >
          <span>
            Target
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="standard"
        key="Standard"
        scope="col"
        tabIndex={0}
      >
        <div
          className="standard"
        >
          <span>
            Standard
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="dok"
        key="DOK"
        scope="col"
        tabIndex={0}
      >
        <div
          className="dok"
        >
          <span>
            DOK
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="difficulty"
        key="Difficulty"
        scope="col"
        tabIndex={0}
      >
        <div
          className="difficulty"
        >
          <span>
            Difficulty
          </span>
        </div>
      </th>
    </React.Fragment>
    <React.Fragment>
      <th
        className="answerkeys"
        key="Answer key"
        scope="col"
        tabIndex={0}
      >
        <div
          className="answerkeys"
        >
          <span>
            Answer key
          </span>
        </div>
      </th>
    </React.Fragment>
    <th
      className="th-arrrow-button"
      colSpan={1}
      scope="col"
    />
  </tr>
</thead>
`;
